### PR TITLE
[WIP] Features/2167 persist filter and filter UI modification

### PR DESCRIPTION
**Pull request by rparse**

### CODE REVIEW

File: lib/satis/menus/menu.rb [hunk 1]
```diff
@@ -3,20 +3,21 @@
...
     class Menu
-      attr_reader :items, :level
-
+      attr_reader :items, :level, :event
       def initialize(*args, **kwargs)
         @options = args.extract_options!
         @items = []
         @scope = Array.wrap(args.first)
         @level = kwargs[:level] || 0
+        @event = "mouseover->satis-menu#show mouseleave->satis-menu#hide"
```
Comment (tdegrunt) [inline lib/satis/menus/menu.rb:12]: Can we make options[:event] either be `mouseover` or `click` perhaps and based on that set @event?

File: app/components/satis/menu_item/component.html.slim [hunk 2]
```diff
@@ -1,9 +1,13 @@
 li.rounded-sm.px-3.py-1.hover:bg-gray-100.dark:hover:bg-gray-200.flex.items-center data-controller="satis-menu" data-action="mouseover->satis-menu#show mouseleave->satis-menu#hide"
   a.cursor-pointer.py-1.w-full.text-left.flex.items-center.outline-none.focus:outline-none href=item.link *item.link_attributes
+    - if item.type == :form
```
Comment (tdegrunt) [inline app/components/satis/menu_item/component.html.slim:3]: Should this than still be named form, or is `:custom` better?
I'd rename file_name to path.

Comment (rparse) [reply app/components/satis/menu_item/component.html.slim:3]: Yes sure, Will change it, Thank you!

File: lib/satis/menus/menu.rb (hunk 1 above)
Comment (rparse) [reply lib/satis/menus/menu.rb:12]: This one I have set as default and dynamic one I have handled at line 20 below, please check if its okay
Thank you!

File: app/components/satis/menu/component.html.slim [hunk 3]
```diff
@@ -1,9 +1,9 @@
-div data-controller="satis-menu" data-action="mouseover->satis-menu#show mouseleave->satis-menu#hide"
+div data-controller="satis-menu" data-action=menu.event
   - if content
     = content
   - else
     button.inline-flex.items-center.justify-center.h-8.w-8.rounded-full.focus:outline-none.focus:ring-2.focus:ring-offset-2.focus:ring-primary-500.dark:text-gray-500 aria-expanded="false" aria-haspopup="true" type="button"
-      span.font-semibold.flex-1 class="#{menu.items.present? ? '' : 'text-gray-200'}"
+      span.font-semibold.flex-1 id="filter_icon#{selected_column}" class="#{menu.items.present? ? '' : 'text-gray-200'}"
```
Comment (tdegrunt) [inline app/components/satis/menu/component.html.slim:6]: What is selected_column in relation to a menu?

File: app/components/satis/menu/component.rb [hunk 4]
```diff
@@ -4,12 +4,13 @@ module Satis
   module Menu
     class Component < Satis::ApplicationComponent
       # renders_many :tabs, Tab::Component
-      attr_reader :menu, :icon
+      attr_reader :menu, :icon, :selected_column
```
Comment (tdegrunt) [inline app/components/satis/menu/component.rb:7]: What is selected_column in relation to a menu?

File: app/components/satis/tabs/component.rb [hunk 5]
```diff
@@ -4,14 +4,26 @@ module Satis
   module Tabs
     class Component < Satis::ApplicationComponent
       renders_many :tabs, Tab::Component
-      attr_reader :group, :persist, :key
+      attr_reader :group, :persist, :action_table_view, :key
```
Comment (tdegrunt) [inline app/components/satis/tabs/component.rb:7]: a tab knows nothing about action-table and shouldn't in my belief.

File: lib/satis/menus/menu.rb (hunk 1 above)
Comment (tdegrunt) [reply lib/satis/menus/menu.rb:12]: Well I meant if a developer would set event to "mouseover" we make the view use "mouseover->satis-menu#show mouseleave->satis-menu#hide", and for click the same. Now you do something similar with @scope.include(:filter_menu).

**Review by tdegrunt — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 6]
```diff
@@ -1,7 +1,3 @@
 Satis::Engine.routes.draw do
-  resources :tables, param: :table_name do
```
Comment (tdegrunt) [inline config/routes.rb:2]: 👍🏻

File: lib/satis/menus/menu.rb (hunk 1 above)
Comment (rparse) [reply lib/satis/menus/menu.rb:12]: Fixed! Thank you.

File: app/components/satis/menu/component.html.slim (hunk 3 above)
Comment (rparse) [reply app/components/satis/menu/component.html.slim:6]: This is for handling filter icon of selected column, we need to morph only the selected column filter so used this.

**Pull request closed without merging**